tests/codeception/acceptance/SubThemeCest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+use Drupal\Core\Serialization\Yaml;
+use Faker\Factory;
+
+/**
+ * Class SubThemeCest.
+ */
+class SubThemeCest {
+
+  /**
+   * Human readable name for the theme.
+   *
+   * @var string
+   */
+  protected $themeName;
+
+  /**
+   * Real path to the theme directory.
+   *
+   * @var string
+   */
+  protected $themePath;
+
+  /**
+   * SubThemeCest constructor.
+   */
+  public function __construct() {
+    $this->themeName = Factory::create()->firstName;
+    $this->themePath = realpath(dirname(drupal_get_path('theme', 'stanford_basic'))) . '/' . strtolower($this->themeName);
+  }
+
+  /**
+   * Before the tests, create a stubbed out subtheme.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   */
+  public function _before(AcceptanceTester $I) {
+    $this->createTheme();
+  }
+
+  /**
+   * Always cleanup the config after testing.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   */
+  public function _after(AcceptanceTester $I) {
+    $this->_failed($I);
+  }
+
+  /**
+   * Always cleanup the config after testing.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   */
+  public function _failed(AcceptanceTester $I) {
+    $this->runConfigImport($I, TRUE);
+    unlink($this->themePath . '/' . strtolower($this->themeName) . '.info.yml');
+    rmdir($this->themePath);
+  }
+
+  /**
+   * Enable the subtheme and the config should reflect the changes done.
+   */
+  public function testSubTheme(AcceptanceTester $I) {
+    $I->runDrush('theme:enable -y ' . strtolower($this->themeName));
+    $I->logInWithRole('administrator');
+    $I->amOnPage('/admin/appearance');
+    $I->click('Set as default', sprintf('a[title="Set %s as default theme"]', $this->themeName));
+    $I->amOnPage('/');
+    $I->canSeeResponseCodeIs(200);
+
+    $this->runConfigImport($I);
+    $result = $I->runDrush('config-get system.theme default --format=json');
+    $result = json_decode($result, TRUE);
+    $I->assertEquals(strtolower($this->themeName), $result['system.theme:default']);
+
+    $I->amOnPage('/admin/appearance');
+    $I->click('Set as default', sprintf('a[title="Set %s as default theme"]', 'Stanford Basic'));
+    $this->runConfigImport($I);
+
+    $result = $I->runDrush('config-get system.theme default --format=json');
+    $result = json_decode($result, TRUE);
+    $I->assertEquals('stanford_basic', $result['system.theme:default']);
+
+    $I->amOnPage('/');
+    $I->canSeeResponseCodeIs(200);
+  }
+
+  /**
+   * Run config import and adjust saml module if necessary.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   * @param bool $disable_config_ignore
+   *   If config ignore module should be disabled first.
+   */
+  protected function runConfigImport(AcceptanceTester $I, $disable_config_ignore = FALSE) {
```

Comment:
I see what you are trying to do here but this could be destructive or disruptive as the codeception tests run in shuffle. This could possibly undo/unset something that was expected. I am not sure how I feel about this yet. Perhaps we can get @imonroe's big brain on this to help think through if this task could cause issues in test running.

Reply:
all tests should assume a default state ootb. So this just resets that. If a module changes a config before this, another test should not be relying on that change

Reply:
Aye, just want a second opinion on that assumption.

Reply:
The theme creation/destruction happens all in this class, so I don't think it'll interfere with anything else; the tests may run in shuffle, but they don't run simultaneously.  I actually think this approach of spinning up a mock theme to test with is pretty smart.  It may be worth abstracting out into a mocker we can re-use later as we proceed with subthemeing and the testing associated with it.

Reply:
Thanks for the feedback